src/MOI_wrapper.jl

```diff
@@ -179,182 +408,1175 @@ function MOI.get(o::Optimizer, param::MOI.RawParameter)
+    return MOI.Utilities.get_fallback(model, attr, Float64)
+end
+
+function MOI.get(model::Optimizer, ::MOI.SolveTime)
+    return Highs_getHighsRunTime(model)
+end
+
+function MOI.get(model::Optimizer, ::MOI.SimplexIterations)
+    return Highs_getSimplexIterationCount(model)
+end
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.VariablePrimal,
+    x::MOI.VariableIndex,
+)
+    MOI.check_result_index_bounds(model, attr)
+    return model.solution.colvalue[column(model, x) + 1]
+end
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintPrimal,
+    c::MOI.ConstraintIndex{MOI.SingleVariable,<:_SCALAR_SETS}
+)
+    MOI.check_result_index_bounds(model, attr)
+    return MOI.get(model, MOI.VariablePrimal(), MOI.VariableIndex(c.value))
+end
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintPrimal,
+    c::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64},<:_SCALAR_SETS},
+)
+    MOI.check_result_index_bounds(model, attr)
+    return model.solution.rowvalue[row(model, c) + 1]
+end
+
+"""
+    _signed_dual(model::Optimizer, dual::Float64, ::Type{Set})
+
+A heuristic for determining whether the dual of an interval constraint applies
+to the lower or upper bound. It can be wrong by at most the solver's tolerance.
+"""
+function _signed_dual end
+
+function _signed_dual(
+    model::Optimizer, dual::Float64, ::Type{MOI.LessThan{Float64}}
+)
+    sense = MOI.get(model, MOI.ObjectiveSense())
+    if sense == MOI.MIN_SENSE && dual < 0
+        return dual
+    elseif sense == MOI.MAX_SENSE && dual > 0
+        return -dual
+    else
+        return 0.0
+    end
+end
+
+function _signed_dual(
+    model::Optimizer, dual::Float64, ::Type{MOI.GreaterThan{Float64}}
+)
+    sense = MOI.get(model, MOI.ObjectiveSense())
+    if sense == MOI.MIN_SENSE && dual > 0
+        return dual
+    elseif sense == MOI.MAX_SENSE && dual < 0
+        return -dual
+    else
+        return 0.0
+    end
+end
+
+function _signed_dual(model::Optimizer, dual::Float64, S::Any)
+    sense = MOI.get(model, MOI.ObjectiveSense())
+    return sense == MOI.MAX_SENSE ? -dual : dual
+end
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintDual,
+    c::MOI.ConstraintIndex{MOI.SingleVariable,S},
+) where {S<:_SCALAR_SETS}
+    MOI.check_result_index_bounds(model, attr)
+    reduced_cost = model.solution.coldual[column(model, c) + 1]
+    return _signed_dual(model, reduced_cost, S)
+end
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintDual,
+    c::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64},S},
+) where {S<:_SCALAR_SETS}
+    MOI.check_result_index_bounds(model, attr)
+    dual = model.solution.rowdual[row(model, c) + 1]
+    # TODO(odow): Ask HiGHS why their convention for row duals is the opposite
+    # of their convention for column duals. I guess because they transform the
+    # constraints into `Ax - Iy = 0`?
+    return _signed_dual(model, -dual, S)
```

Comment:
@galabovaa is there a document on your duality convention?

Reply:
The HiGHS row duals are right if you transform the constraints into Ax + Iy = 0 and flip+negate the bounds on the constraints to get bounds on y - which is what the simplex solver does. We did this for consistency with something, but I don't remember what: possibly OSL. My instinct is not to have a different convention, so there must have been a good reason! Clearly we can't change now. We should document it, though

Reply:
Thanks for the explanation. And definitely no need to change. It's all rather arbitrary. I always get a little confused, because JuMP uses conic duality instead of LP. So the dual of a `>=` is always non-negative, and the dual of a `<=` is always non-positive, regardless of objective sense.